nmdc_schema/migrators/partials/migrator_from_10_2_0_to_11_0_0/migrator_from_10_2_0_to_11_0_0_part_17.py

```diff
@@ -0,0 +1,50 @@
+from nmdc_schema.migrators.migrator_base import MigratorBase
+
+
+class Migrator(MigratorBase):
+    r"""
+    Migrates a database between two schemas.
+    """
+
+    _from_version = "XX"
+    _to_version = "PR250"
+
+    def upgrade(self) -> None:
+        r"""
+        Migrates the database from conforming to the original schema, to conforming to the new schema.
+
+        Should be run after migrator_from_10_2_0_to_11_0_0_part_16.py.
+        """
+        # Add a type slot on the Protocol class within the protocol_link slot on each document
+        collections_to_update = [
+            "data_generation_set",
+            "material_processing_set",
+            "collecting_biosamples_from_site_set",
+            "protocol_execution_set",
+            "storage_process_set",
+            "workflow_execution_set",
+            "study_set"
+        ] 
+
+        for collection_name in collections_to_update:
+            self.adapter.process_each_document(collection_name, [self.add_type_to_protocol_link])
+    
+    def add_type_to_protocol_link(self, document: dict) -> dict:
+        r"""
+        Add a type slot on the Protocol class within the protocol_link slot on each document
```

Comment:
In the context of Mongo documents (as opposed to the schema), I prefer "field" instead of "slot".

```suggestion
        Add a `type` field to the `Protocol` instance within the `protocol_link` field on each document.
```